### PR TITLE
Make `CairoRunner::write_output()` work with unordered builtins.

### PR DESCRIPTION
**Pull request by azteca1998**
# Make `CairoRunner::write_output()` work with unordered builtins.

## Description

  - Make `CairoRunner::write_output()` work with unordered builtins.
  - Remove safe (but unnecessary) `.unwrap()`.
  - Test the changes.

## Checklist
- [ ] Linked to Github Issue
- [x] Unit tests added
- [ ] Integration tests added.
- [x] This change requires new documentation.
  - [x] Documentation has been added/updated.

### CODE REVIEW

**Comment by azteca1998** [comment]
[Fix output builtin lookup in CairoRunner::write_output()](https://trello.com/c/2yd9BLvF/135-fix-output-builtin-lookup-in-cairorunnerwriteoutput)

File: src/vm/runners/cairo_runner.rs [hunk 1]
```diff
@@ -824,43 +824,47 @@ impl CairoRunner {
...
-                    stdout,
-                    "{}",
-                    to_field_element(value.into_owned(), vm.prime.clone())
-                )
-                .map_err(|_| RunnerError::WriteFail)?;
-            }
+        vm.segments.compute_effective_sizes(&vm.memory);
+        let base = builtin.base();
+
+        let segment_used_sizes = vm.segments.compute_effective_sizes(&vm.memory);
```
Comment (fmoletta) [inline src/vm/runners/cairo_runner.rs:849]: Why is compute_effective_sizes called twice?

File: src/vm/vm_memory/memory_segments.rs [hunk 2]
```diff
@@ -68,16 +68,10 @@ impl MemorySegmentManager {
         }
     }
 
-    ///Calculates the size (number of non-none elements) of each memory segment
-    pub fn compute_effective_sizes(&mut self, memory: &Memory) {
-        if self.segment_used_sizes != None {
-            return;
-        }
```
Comment (fmoletta) [inline src/vm/vm_memory/memory_segments.rs:75]: Why was this removed?

Comment (azteca1998) [reply src/vm/vm_memory/memory_segments.rs:75]: `Option::get_or_insert_with` will not call the closure if it's already present, but will return a reference instead. It's the same behaviour.

File: src/vm/runners/cairo_runner.rs (hunk 1 above)
Comment (azteca1998) [reply src/vm/runners/cairo_runner.rs]: It appears I might have copy-pasted instead of cut-pasted. Fixed.

File: src/vm/vm_memory/memory_segments.rs (hunk 2 above)
Comment (fmoletta) [reply src/vm/vm_memory/memory_segments.rs:75]: Sounds good!

**Review by fmoletta — APPROVED** [review]

**Review by Jrigada — APPROVED** [review]
